src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.LocalFunctions.cs

```diff
@@ -0,0 +1,301 @@
+                                                         capturedSymbol: capturedSymbol,
+                                                         slot: slot);
+            if (!info.UsedVariables.Contains(usedVar))
+            {
+                info.UsedVariables.Add(usedVar);
+                info.IsDirty = true;
+            }
+        }
+
+        private void VisitLocalFunctionAccess(
+            LocalFunctionSymbol localFunc,
+            CSharpSyntaxNode syntax,
+            bool write)
+        {
+            _usedLocalFunctions.Add(localFunc);
+            LocalFuncInfo info;
+            if (LocalFuncResults.TryGetValue(localFunc, out info))
+            {
+                foreach (var usedVar in info.UsedVariables)
+                {
+                    bool capturedInLocalFunction =
+                        IsCapturedInLocalFunction(usedVar.CapturedSymbol);
+
+                    if (write && usedVar.Written)
+                    {
+                        if (capturedInLocalFunction)
+                        {
+                            AssignVarInLocalFunction(usedVar.SlotOpt.Value, usedVar.CapturedSymbol);
+                        }
+                        else
+                        {
+                            SetSlotState(usedVar.SlotOpt.Value, assigned: true);
+                        }
+                    }
+                    else
+                    {
+                        if (capturedInLocalFunction)
+                        {
+                            NoteReadInLocalFunction(usedVar.SymbolOpt,
+                                                    usedVar.CapturedSymbol,
+                                                    slotOpt: usedVar.SlotOpt);
+                        }
+                        else
+                        {
+                            CheckAssigned(usedVar.SymbolOpt, syntax, usedVar.SlotOpt);
+                        }
+                    }
+                }
+            }
+
+            var containingLocalFunc = GetNearestLocalFunctionOpt();
+            if ((object)containingLocalFunc != null)
+            {
+                info = LocalFuncResults[containingLocalFunc];
+                if (!info.UsedLocalFunctions.Contains(localFunc))
+                {
+                    info.UsedLocalFunctions.Add(localFunc);
+                    info.IsDirty = true;
+                }
+            }
+        }
+
+        private bool IsCapturedInLocalFunction(Symbol variable,
+            ParameterSymbol rangeVariableUnderlyingParameter = null)
+        {
+            if (variable == null)
+            {
+                return false;
+            }
+            
+            if (!IsCaptured(variable,
+                            rangeVariableUnderlyingParameter,
+                            currentMethodOrLambda))
+            {
+                return false;
+            }
+
+            var localFunc = GetNearestLocalFunctionOpt();
```

Comment:
That may return the lambda. However, if we read a variable captured inside a lambda inside a local function, we need to make sure we defer reporting until the local function is used (since the lambda's read-point doesn't happen until the local function is used).